crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -402,6 +378,34 @@ impl Rules {
 
         Ok(commands)
     }
+
+    /// Apply a list of command to the managed ruleset.
+    ///
+    /// The commands may silently fail because the ruleset may have changed
+    /// between the time the commands were created and the time it is applied.
+    pub(crate) fn apply_commands(&mut self, commands: &[Command]) {
```

Comment:
This should take ownership of the `Vec<Command>`, unless we have a (non-test only) reason to reuse them after applying.